crates/utils/src/utils/validation.rs

```diff
@@ -257,16 +255,23 @@ pub fn build_and_check_regex(regex_str_opt: &Option<&str>) -> LemmyResult<Option
   )
 }
 
-pub fn clean_url_params(url: &Url) -> Url {
-  let mut url_out = url.clone();
-  if let Some(query) = url.query() {
-    let new_query = query
-      .split_inclusive('&')
-      .filter(|q| !CLEAN_URL_PARAMS_REGEX.is_match(q))
-      .collect::<String>();
-    url_out.set_query(Some(&new_query));
+/// Cleans a url of tracking parameters.
+pub fn clean_url(url: &Url) -> LemmyResult<Url> {
+  match URL_CLEANER.clear_single_url(url.as_str()) {
+    Ok(res) => Ok(Url::parse(&res)?),
```

Comment:
I'd rather not have to return a result, but since this goes from url -> str -> url , it means I have to parse the link again. 

I could run `expect` or `unwrap_or` , and just assume the crate is url-safe strings tho.

Reply:
There is a method [clear_single_url](https://docs.rs/clearurls/latest/clearurls/struct.UrlCleaner.html#method.clear_single_url) but strangely it returns a str as well.

@jendrikw Any chance you can change the above method to return Url instead?

Reply:
I'll add an issue over there, about Url support instead of strings.

Reply:
released in 0.0.3